content/blog/2023-06-25-intruducing-continuous-profiling-skywalking-with-ebpf/index.md

```diff
@@ -0,0 +1,232 @@
+---
+title: "Activating Performance Analysis Tools with SkyWalking and eBPF under Threshold Conditions -- Continuous Profiling"
+date: 2023-06-25
+author: "Han Liu"
+description: "Introduce and demonstrate how SkyWalking and eBPF implement process monitoring with low energy consumption. It will also detail how, when certain conditions are met, these tools initiate more complex performance tasks to monitor programs."
+tags:
+- eBPF
+- Profiling
+- Tracing
+---
+
+# Background
+
+In previous articles, We have discussed how to use SkyWalking and eBPF for performance problem detection within [processes](/blog/2022-07-05-pinpoint-service-mesh-critical-performance-impact-by-using-ebpf) and [networks](blog/diagnose-service-mesh-network-performance-with-ebpf). 
+However, there are still two outstanding issues:
+
+1. **Task initiation time**: It's often challenging to quickly determine which processes require performance monitoring when problems occur. 
+Typically, manual intervention is required to identify processes and the types of performance analysis necessary. 
+We need a faster way to detect which processes require performance analysis.
+2. **Resource consumption of tasks**: When tasks are initiated, they may consume more system resources to analyze more extensive performance issues. 
+We need a method to manage resource consumption and understand which processes necessitate performance analysis.
```

Comment:
This seems not a reason for continuous profiling. We are costing the same resources, or even more?

Do you want to say, continuous profiling would allow the profiling activated by default from the more experienced and senior SRE and Operation engineer?

Reply:
My point is, if performance tasks are continuously started for a particular process by default, it might consume a significant amount of system resources, affecting overall system performance. Continuous analysis can help mitigate this resource consumption

Reply:
OK, then, I think there are 3 points 
1. Current one.
2. The difficulties to determine the profiling scope. Wider profiling causes more resources than it should. (your point 2).
3. Engineer capabilities. On-call is usually covered by the whole team, which have junior and senior engineers, even senior engineers have their understanding limitation of the complex distributed system, it is nearly impossible to understand the whole system by a single one person.

Reply:
updated, please re-check.